plugins/hls-semantic-tokens-plugin/src/Ide/Plugin/SemanticTokens/Internal.hs

```diff
@@ -98,9 +131,6 @@ getSemanticTokensRule recorder =
     let hsFinder = idSemantic getTyThingMap (hieKindFunMasksKind hieKind) refMap
     return $ computeRangeHsSemanticTokenTypeList hsFinder virtualFile ast
 
--- | Persistent rule to ensure that semantic tokens doesn't block on startup
```

Comment:
why drop this?

Reply:
Hmm, so does this mean we have to wait quite a while to get responses when we start up?

Reply:
yes, as we have been. Persistent rule won't help us here(Since it is not `ideAction`, the persistent rule serves only as a fall back when the actual computation of the rule failed and not already having a cache) .